Mtd 49 new methods connected to TravelDiary entity

ghilyana opened the pull request:
<3

ghilyana committed:
The DAO and DTO are created for the TravelDiary

ghilyana committed:
The DAO and DTO are created for the TravelDiary; the method create is added

ghilyana committed:
Methods to check before saving and tests are rewritten.

ghilyana committed:
Merge branch 'master' into MTD-43_create

ghilyana committed:
Methods to check before saving and tests are rewritten.

ghilyana committed:
Problems btw CityDAO and CountryDAO are fixed.

ghilyana committed:
Problems btw CityDAO and CountryDAO are fixed.

ghilyana committed:
Merge branch 'master' into MTD-50_TravelDiaryDAOandDTO

# Conflicts:
#	src/main/java/com/ghilly/model/DAO/CityDAO.java

ghilyana committed:
TravelDiaryDAO is created

ghilyana committed:
Merge branch 'master' into MTD-43_create

# Conflicts:
#	src/main/java/com/ghilly/model/DAO/CityDAO.java
#	src/main/java/com/ghilly/model/DAO/CountryDAO.java
#	src/main/java/com/ghilly/model/DAO/TravelDiaryDAO.java
#	src/main/java/com/ghilly/model/DTO/TravelDiaryDTO.java
#	src/main/java/com/ghilly/transformer/TransformerFromDAOtoDTOAndBack.java
#	src/main/java/com/ghilly/web/controller/CityController.java
#	src/main/java/com/ghilly/web/controller/CountryController.java
#	src/test/java/com/ghilly/web/controller/CountryControllerTest.java

ghilyana committed:
commit before weekend

ghilyana committed:
commit before weekend

ghilyana committed:
good version with tx manager

ghilyana committed:
good version

ghilyana committed:
good version after refactoring dao folder and some places in classes connected to the entities of this folder

ghilyana committed:
good version after refactoring dto folder and some places in classes connected to the entities of this folder

ghilyana committed:
good version after refactoring service and controller folders and some places in classes connected to the entities of this folder

ghilyana committed:
last version with successful tests

ghilyana committed:
commit before unstash

ghilyana committed:
commit for working version, method getById

ghilyana committed:
commit for working version, method getById, getAll and delete

gregorusanov commented on src/main/java/com/ghilly/model/dto/TravelDiary.java:
```diff
@@ -0,0 +1,30 @@
+    private int rating;
+    private Set<Integer> cityIdSet = new HashSet<>();
+
+    public TravelDiary(int id, String arrivalDate, String departureDate) {
```
У тебя же есть билдер, зачем отдельный конструктор?

ghilyana replied:
Исправила, спасибо.

ghilyana replied:
Исправила, спасибо.

ghilyana replied:
Спасибо, исправила.

gregorusanov commented on src/main/java/com/ghilly/model/dto/Country.java:
```diff
@@ -0,0 +1,19 @@
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
```
здесь билдер не нужен, всего два параметра

ghilyana replied:
Убрала, спасибо.

ghilyana replied:
Исправила, спасибо.

ghilyana replied:
Спасибо, исправила.

gregorusanov commented on src/main/java/com/ghilly/model/dto/City.java:
```diff
@@ -0,0 +1,32 @@
+        this.name = name;
+    }
+
+    public City(String name, int countryId, boolean capital) {
```
я не совсем понимаю, зачем тебе конструкторы если есть билдер

ghilyana replied:
Исправила, спасибо.

ghilyana replied:
Спасибо, исправила.

gregorusanov commented on src/main/java/com/ghilly/model/dao/CityEntity.java:
```diff
@@ -0,0 +1,72 @@
+    @EqualsAndHashCode.Exclude
+    private Set<CityTravelDiaryEntity> travelDiaryCitySet = new HashSet<>();
+
+    public CityEntity(int id, String name) {
```
Зачем столько конструкторов, если у тебя есть билдер?

ghilyana replied:
Спасибо, исправила.

gregorusanov commented on src/main/java/com/ghilly/model/dao/CountryEntity.java:
```diff
@@ -0,0 +1,40 @@
+    private Set<CityEntity> citySet = new HashSet<>();
+
+
+    public CountryEntity(int id, String name) {
```
у тебя же есть AllArgsConstructor, можно не определять свой конструктор

ghilyana replied:
Спасибо, исправила.

gregorusanov commented on src/main/java/com/ghilly/web/ErrorHandler.java:
```diff
@@ -51,4 +51,32 @@ public ResponseEntity<String> catchCityAlreadyExistsException(CityAlreadyExistsE
                 .body(exception.getMessage());
     }
+
+    @ExceptionHandler(IllegalDateException.class)
```
у тебя одинаковый код для  всех ошибок, лучше всего поместить в массив {} в анотации
({IllegalDateException.class,  IllegalBudgetException.class, ...})

ghilyana replied:
Спасибо, исправила.

gregorusanov commented on src/main/java/com/ghilly/service/TravelDiaryServiceRest.java:
```diff
@@ -0,0 +1,53 @@
+    private final TravelDiaryRepository travelDiaryRepository;
+
+    public TravelDiaryServiceRest(CityRepository cityRepository, TravelDiaryRepository travelDiaryRepository) {
+        this.cityRepository = cityRepository;
```
нигде не используется

ghilyana replied:
Используется в конфигурации.

ghilyana committed:
commit for working version, method getById, getAll and delete;
there were changes in entity classes and test classes: a) builder is deleted from Country and CountryEntity classes;
          b) constructors are removed from the classes where @builder was mentioned;
          c) two last changes caused problems in tests, so they are rewritten.

ghilyana committed:
commit for working version, method getById, getAll and delete;
there were changes in entity classes and test classes: a) builder is deleted from Country and CountryEntity classes;
          b) constructors are removed from the classes where @builder was mentioned;
          c) two last changes caused problems in tests, so they are rewritten.
